From 0bff1b4fc6929694b706bd7e659ed0b58ff51f2d Mon Sep 17 00:00:00 2001
From: Liam Middlebrook <lmiddlebrook@nvidia.com>
Date: Tue, 11 May 2021 20:23:18 -0700
Subject: [PATCH] loader: Set default regkey for NVIDIA NGX FullPath

Sets the default location for the NVIDIA NGX SDK search-path to be
C:\Windows\System32\

This is required for supporting NVIDIA DLSS within Proton.

Reviewed-by: Adam Moss <amoss@nvidia.com>
---
 loader/wine.inf.in | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 851a31b7d05..735a3899b3e 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -75,7 +75,8 @@ AddReg=\
     Timezones,\
     VersionInfo,\
     LicenseInformation, \
-    SteamClient
+    SteamClient, \
+    NVIDIANGX
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -104,7 +105,8 @@ AddReg=\
     Timezones,\
     VersionInfo.ntamd64,\
     LicenseInformation, \
-    SteamClient.ntamd64
+    SteamClient.ntamd64, \
+    NVIDIANGX
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -149,7 +151,8 @@ AddReg=\
     Tapi,\
     VersionInfo.ntamd64,\
     LicenseInformation, \
-    SteamClient.ntamd64
+    SteamClient.ntamd64, \
+    NVIDIANGX
 
 [Wow64Install.ntarm64]
 WineFakeDlls=FakeDllsWin32
@@ -4431,6 +4431,9 @@ HKCU,Software\Wine\DllOverrides,"vcomp140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"atiadlxx",,"disabled"
 
+[NVIDIANGX]
+HKLM,Software\NVIDIA Corporation\Global\NGXCore,"FullPath",,"C:\Windows\System32"
+
 [NlsFiles]
 c_037.nls
 c_10000.nls
From e3d19be8982e063183ff6ffafb96045c8d7a38fb Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Fri, 10 Sep 2021 23:16:57 +0300
Subject: [PATCH] wine.inf: Disable nvcuda.dll by default.

The nvcuda.dll stub presence may cause regressions when the games
successfully load and try to use it (like Divinity The Original Sin 2).
The library is enabled in proton script together with nvapi.dll.

Link: https://github.com/ValveSoftware/wine/pull/119
---
 loader/wine.inf.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index dd2be6a2a45..644a2fcce1d 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -4208,6 +4208,7 @@ HKCU,Software\Wine\DllOverrides,"ucrtbase",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcomp140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"atiadlxx",,"disabled"
+HKCU,Software\Wine\DllOverrides,"nvcuda",0x2,"disabled"

 [SteamClient.ntamd64]
 HKCU,Software\Valve\Steam,"SteamPath",,"%16422%\Steam"
@@ -4263,6 +4264,7 @@ HKCU,Software\Wine\DllOverrides,"ucrtbase",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcomp140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
 HKCU,Software\Wine\DllOverrides,"atiadlxx",,"disabled"
+HKCU,Software\Wine\DllOverrides,"nvcuda",0x2,"disabled"

 [NVIDIANGX]
 HKLM,Software\NVIDIA Corporation\Global\NGXCore,"FullPath",,"C:\Windows\System32"
